internal/pkg/service/stream/encryption/encryption.go

```diff
@@ -0,0 +1,72 @@
+package encryption
+
+import (
+	"context"
+	"crypto/rand"
+	"math/big"
+
+	"github.com/keboola/go-cloud-encrypt/pkg/cloudencrypt"
+)
+
+const (
+	ProviderNative = Provider("native")
+	ProviderGCP    = Provider("gcp")
+	ProviderAWS    = Provider("aws")
+	ProviderAzure  = Provider("azure")
+)
+
+type Provider string
+
+func NewEncryptor(ctx context.Context, config Config) (cloudencrypt.Encryptor, error) {
+	var encryptor cloudencrypt.Encryptor
+	var err error
+
+	switch config.Provider {
+	case ProviderNative:
+		encryptor, err = cloudencrypt.NewNativeEncryptor([]byte(config.Native.SecretKey))
+		if err != nil {
+			return nil, err
+		}
+	case ProviderGCP:
+		encryptor, err = cloudencrypt.NewGCPEncryptor(ctx, config.GCP.KMSKeyID)
+		if err != nil {
+			return nil, err
+		}
+	case ProviderAWS:
+		encryptor, err = cloudencrypt.NewAWSEncryptor(ctx, config.AWS.Region, config.AWS.KMSKeyID)
+		if err != nil {
+			return nil, err
+		}
+	case ProviderAzure:
+		encryptor, err = cloudencrypt.NewAzureEncryptor(ctx, config.Azure.KeyVaultURL, config.Azure.KeyName)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if config.Provider != ProviderNative {
+		encryptor, err = cloudencrypt.NewDualEncryptor(ctx, encryptor)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return encryptor, nil
+}
+
+const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+func RandomSecretKey() (string, error) {
```

Comment:
This I would put under `utils/encryption`

Reply:
It's fine here in my opinion. If anything it could be moved into tests.

Reply:
This is pure utility/separate function. Check for example `idgenerator`. I would like to split generic randoms in each module to have it together so we can reuse it. I would be beneficial for us to prevent code duplication and having random strings, numbers etc thrown over monorepository

Reply:
Removed. https://github.com/keboola/keboola-as-code/commit/e90d3cbb5b6b5978e8ca702aa11f0f6b43b12aa9